fdmi/fol_fdmi_src.c

```diff
@@ -108,6 +108,96 @@
  * rh_lsn_discarded for it's BE domain could be discarded from deduplication
  * persistence because there is nothing in the cluster that is going to send
  * FDMI FOL record about this operation.
+ *
+ * @section FDMI FOL records resend
+ *
+ * There are 2 major cases here:
+ *
+ * 1. FDMI plugin restarts, FDMI source is not. In this case FDMI source needs
+ *    to resend everything FDMI plugin hasn't confirmed consumption for. This
+ *    could be done by FDMI source dock fom by indefinitely resending FDMI
+ *    records until either FDMI plugin confirms consumption or FDMI plugin
+ *    process fails permanently.
+ * 2. FDMI source restarts. The following description is about this case.
+ *
+ * FDMI source may restart unexpectedly (crash/restart) or it might restart
+ * gracefully (graceful shutdown/startup). In either case there may be FDMI FOL
+ * records that don't have consumption confirmation from FDMI plugin. Current
+ * implementation takes BE tx reference until there is such confirmation from
+ * FDMI plugin. BE tx reference taken means ih this case that BE tx wouldn't be
+ * discarded from BE log until the reference is put. BE recovery recovers all
+ * transactions that were not discarded, which is very useful for FDMI FOL
+ * record resend case: if all FOL records for such recovered transactions are
+ * resent as FDMI FOL records then it will be no FDMI FOL record missing on FDMI
+ * plugin side regardless of if FDMI source restarts or not and how and how many
```

Comment:
s/regardless of if/regardless whether

s/and how and how many/, how and how many

Reply:
Fixed.